sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala

```diff
@@ -82,3 +82,22 @@ abstract class UserDefinedType[UserType] extends DataType with Serializable {
   override private[sql] def acceptsType(dataType: DataType) =
     this.getClass == dataType.getClass
 }
+
+/**
+ * ::DeveloperApi::
+ * The user defined type in Python.
+ *
+ * Note: This can only be accessed via Python UDF, or accessed as serialized object.
+ */
+class PythonUserDefinedType(val sqlType: DataType, pyClass: String) extends UserDefinedType[Any] {
+  override def pyUDT = pyClass
+  def serialize(obj: Any): Any = obj
```

Comment:
are these interfaces designed in UserDefinedType? If yes, can you add override here? it caches a lot of potential problems when we do refactoring down the line.